src/main/shared/store/store.ts

```diff
@@ -0,0 +1,85 @@
+import { StoreData, StoreEntity } from './store-model';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import * as deepmerge from 'deepmerge';
+import { filter, map } from 'rxjs/operators';
+import { Utils } from '../util/utils';
+import { IObject } from '../util/util-model';
+
+export abstract class Store<T extends IObject = IObject> {
```

Comment:
Are you ok with this kind of data store? It is necessary to keep track of two types of updates like update (comes from the network) and commit (comes from locally), because some business model like the client player can be update both from the network (server says you are at the position xy) and from the client (you are moving) and need to prevent infinite loop.

Reply:
yeah seems nice

so one store per "room"?

Reply:
Yes that would be a good solution, however, there is only one store yet.